packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -375,15 +170,6 @@ export function addShadowRootEventListener(
     listener: EventListener,
     _options?: boolean | AddEventListenerOptions
 ) {
-    if (process.env.NODE_ENV !== 'production') {
-        if (!isFunction(listener)) {
-            throw new TypeError(
-                `Invalid second argument for ShadowRoot.addEventListener() in ${toString(
-                    sr
-                )} for event "${type}". Expected an EventListener but received ${listener}.`
-            );
-        }
-    }
```

Comment:
How come we are removing this guardrail?

Reply:
same as above... moved up one layer.